XLS-0081d-permissioned-dex/README.md

```diff
@@ -0,0 +1,593 @@
+|`Flags`|✔️|`number`|`UInt32`|A bit-map of boolean flags enabled for this object. Currently, the protocol defines no flags for `DirectoryNode` objects. The value is always `0`.|
+|`Indexes`|✔️|`array`|`Vector256`|The contents of this directory: an array of IDs of other objects.|
+|`IndexNext`| |`number`|`UInt64`|If this directory consists of multiple pages, this ID links to the next object in the chain, wrapping around at the end.|
+|`IndexPrevious`| |`number`|`UInt64`|If this directory consists of multiple pages, this ID links to the previous object in the chain, wrapping around at the beginning.|
+|`LedgerEntryType`|✔️|`string`|`UInt16`|The value `0x0064`, mapped to the string `DirectoryNode`, indicates that this object is part of a directory.|
+|`RootIndex`|✔️|`string`|`Hash256`|The ID of root object for this directory.|
+|`TakerGetsCurrency`|✔️|`string`|`Hash160`|The currency code of the `TakerGets` amount from the offers in this directory.|
+|`TakerGetsIssuer`|✔️|`string`|`Hash160`|The issuer of the `TakerGets` amount from the offers in this directory.|
+|`TakerPaysCurrency`|✔️|`string`|`Hash160`|The currency code of the `TakerPays` amount from the offers in this directory.|
+|`TakerPaysIssuer`|✔️|`string`|`Hash160`|The issuer of the `TakerPays` amount from the offers in this directory.|
+</details>
+
+This spec proposes these additions:
+
+| Field Name | Required? | JSON Type | Internal Type | Description |
+|------------|-----------|-----------|---------------|-------------|
+|`DomainID`| |`string`|`Hash256`|The domain that the offer directory is a part of.|
+
+#### 3.1.1. `DomainID`
+
+A domain has its own set of orderbooks.
+
+### 3.2. Object ID
+
+_The proposed modification is in bold._
+
+There are three different formulas for creating the ID of a DirectoryNode, depending on which of the following the DirectoryNode represents:
+
+* The first page (also called the root) of an Owner or NFT Offer directory
+* The first page of an Offer directory
+* Later pages of either type
+
+The first page of an Offer directory has a special ID: the higher 192 bits define the order book, and the remaining 64 bits define the exchange rate of the offers in that directory. (The ID is big-endian, so the book is in the more significant bits, which come first, and the quality is in the less significant bits which come last.) This provides a way to iterate through an order book from best offers to worst. Specifically: the first 192 bits are the first 192 bits of the [SHA-512Half](https://xrpl.org/docs/references/protocol/data-types/basic-data-types#hashes) of the following values, concatenated in order:
+
+* The Book directory space key (`0x0042`)
+* The 160-bit currency code from the `TakerPaysCurrency`
+* The 160-bit currency code from the `TakerGetsCurrency`
+* The AccountID from the `TakerPaysIssuer`
+* The AccountID from the `TakerGetsIssuer`
+* **The `DomainID` of the orderbook, if applicable**
```

Comment:
just want to confirm, the use i can think of this is to allow iteration of offers for particular domainID, which is going to be used by cross currency payment and liquidity calculation in pathfinding

Reply:
Yes. The book directories need to exist, and have distinct hashes from the unpermissioned directories, so what makes the most sense in differentiating those is the DomainID.